packages/forms/src/directives/reactive_errors.ts

```diff
@@ -13,15 +13,26 @@ import {RuntimeErrorCode} from '../errors';
 import {formArrayNameExample, formControlNameExample, formGroupNameExample, ngModelGroupExample} from './error_examples';
 
 
-export function controlParentException(): Error {
+export function controlParentException(controlName: string|number|null): Error {
   return new RuntimeError(
       RuntimeErrorCode.FORM_CONTROL_NAME_MISSING_PARENT,
-      `formControlName must be used with a parent formGroup directive.  You'll want to add a formGroup
+      `formControlName must be used with a parent formGroup directive. You'll want to add a formGroup
       directive and pass it an existing FormGroup instance (you can create one in your class).
 
+      ${describeFormControl(controlName)}
+
     Example:
 
-    ${formControlNameExample}`);
+    ${formControlNameExample}`,
+  );
+}
+
+function describeFormControl(controlName: string|number|null): string {
+  if (controlName) {
+    return `Affected Form Control (name or index): "${controlName}"`;
+  }
```

Comment:
We should update the condition to make sure that we display extra info for `0` as a value. Can we also infer whether it's an index or a name based on the value type? 
```suggestion
  if (controlName != null /* checks both `null` and `undefined` */) {
    return `Affected Form Control ${typeof controlName === 'string' ? 'name' : 'index'}): "${controlName}"`;
  }
```

Reply:
done, hopefully ok